typing/types.ml

```diff
@@ -418,6 +418,28 @@ let equal_tag t1 t2 =
   | Extension (path1,_), Extension (path2,_) -> Path.same path1 path2
   | (Ordinary _ | Extension _), _ -> false
 
+let equal_variant_representation r1 r2 = match r1, r2 with
```

Comment:
Worth shortcutting with `==`? (Honest question -- not sure what best practice is.)